girder_worker_utils/decorators.py

```diff
@@ -16,13 +17,190 @@ class MissingInputException(Exception):
 
 def get_description_attribute(func):
     """Get the private description attribute from a function."""
-    func = getattr(func, 'run', func)
-    description = getattr(func, '_girder_description', None)
+    # func = getattr(func, 'run', func)
+    description = getattr(func, GWFuncDesc._func_desc_attr, None)
     if description is None:
         raise MissingDescriptionException('Function is missing description decorators')
     return description
 
 
+class Argument(object):
+    def __init__(self, name, **kwargs):
+        self.name = name
+        for k, v in six.iteritems(kwargs):
+            setattr(self, k, v)
+
+# No default value for this argument
+class Arg(Argument): pass
+# Has a default argument for the value
+class KWArg(Argument):  pass
+class Varargs(Argument): pass
+class Kwargs(Argument): pass
+# class Return(Argument): pass
```

Comment:
I agree the names could be better.  I'd like to keep support for varargs and kwargs,  at least for CLI they have natural mappings to click's `nargs=-1`  and [forwarding unknown options](https://click.palletsprojects.com/en/5.x/advanced/#forwarding-unknown-options)

Reply:
I'd request at least renaming `KWArg` to `Kwarg` for consistency.

Reply:
yeah this is why it's unfortunately confusing,    KWArg is captialized because it stands for "Key Word Arg"  (e.g  `def foo(key=None)` - 'key' is a key word argument)  which is different then what the inspect module referes too as a kwargs argument (e.g. `def foo(**kwargs)`)

Reply:
I've renamed these to PositionalArg, KeywordArg, VarsArg, KwargsArg